e2e/custom-test-sequencer/testSequencer.js

```diff
@@ -0,0 +1,55 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+const Sequencer = require('@jest/test-sequencer').default;
+const path = require('path');
+
+const priorityOrder = [
+  path.resolve('./d.test.js'),
+  path.resolve('./b.test.js'),
+  path.resolve('./c.test.js'),
+];
+const FAIL = 0;
+
+class CustomSequencer extends Sequencer {
+  sort(tests) {
+    const stats = {};
+    const fileSize = ({path, context: {hasteFS}}) =>
+      stats[path] || (stats[path] = hasteFS.getSize(path) || 0);
+    const hasFailed = (cache, test) =>
+      cache[test.path] && cache[test.path][0] === FAIL;
+    const time = (cache, test) => cache[test.path] && cache[test.path][1];
+
+    tests.forEach(test => (test.duration = time(this._getCache(test), test)));
+    return tests.sort((testA, testB) => {
```

Comment:
Not really related to this review, but note that `sort` sorts in place, thus mutating the array. You should make a copy first (`.slice(0)` or `Array.from`).